portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/locale/Locale.testcase

```diff
@@ -945,4 +945,52 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-157252."
+	@priority = "3"
+	test VirtualHostLanguageCanBeChangedWithUserDefaultLanguage {
+		property test.name.skip.portal.instance = "Locale#VirtualHostLanguageCanBeChangedWithUserDefaultLanguage";
+
+		User.firstLoginPG();
+		
+		JSONLayout.addPublicLayout(
+			groupName = "Test Site Name",
+			layoutName = "Test Page");
+
+		JSONLayout.addWidgetToPublicLayout(
+			groupName = "Test Site Name",
+			layoutName = "Test Page",
+			widgetName = "Asset Publisher");
+
+		Site.openSiteSettingsAdmin(siteURLKey = "test-site-name");
+
+		Site.configureCurrentLanguagesCP(
+			currentSiteLanguages = "English (United States),Spanish (Spain)",
+			defaultSiteLanguage = "Spanish (Spain)");
+
+		Site.openSiteSettingsAdmin(siteURLKey = "test-site-name");
+
+		Site.addVirtualHostsURLCP(
+			pageVirtualHosts = "true",
+			pageVirtualHostURL = "www.able.com");
+
+		Site.openSiteSettingsAdmin(siteURLKey = "test-site-name");
+
+		Site.editVirtualHostLanguagueCP(languageVirtualHost = "Spanish (Spain)");
+
+		Navigator.openSpecificURL(url = "http://www.able.com:8080");
+
+		User.firstLoginUI(specificURL = "http://www.able.com:8080");
+
+		Click(
```

Comment:
We should do some text assert that the Page is being actually displayed in English afterwords, I believe we only check if a locator is present afterwards, or does that locator implicitly assert the text "Control Panel"?

Reply:
the locator implicitly assert the text "Control Panel", I put this one because was the mentioned in the bug